contributor/04-mechanics-of-contributing.md

```diff
@@ -0,0 +1,125 @@
+Excellent!
+Wouldn't it be horrible if all your work would be wasted because the host team is maybe already building something very similar, is planning to depreciate the part of the software you are trying to extend or even does not see what you are proposing to be a fit for their project? 
+This challenge is a frequent challenge and many team relationships suffered from not handling the fit in advance.
+
+Make yourself and the host team happy (and possibly even save some work) by getting agreement from the host team on the user/technical design of the contribution _before_ submitting a pull request.
+You'll have to understand how the host team would like you to reach out for this - sometimes they have regular design discussions in person or via chat/voice/video conferences, some hand in design discussion / extension proposal documents for collective discussion and review, some work with feature requests, etc. 
+Ask a trusted committer about how to best discuss your proposal.  
+
+This type of high-level, up-front agreement early will save time in rework or rejection of your pull request down the road.
+
+## Creating the pull request
+
+### Communication and unblocking yourself
+
+Great, you've made yourself familiar with the host teams approach and they are looking forward to receive your pull request.
+Which gotchas are there waiting for you now?
+
+First, you'll be in less direct contact with them and second you aren't expected to be as knowledgeable and proficient as you might be on the full-time projects that your team owns.
+How can you now deal with this the best?
+
+Try to peruse their documentation, the conversation archives and code artifacts from the host team to unblock yourself - this is pretty much similar to the situation you and likely most people find yourself in when using one of the popular OSS projects.
+
+However, this time you've got the advantage to be able to directly ask the host team if, even after trying to unblock yourself, things went nowhere. (That's an approach that works with many OSS projects as well!)
+It doesn't do either side any good for you to spin forever on a problem that could be resolved in a few-minute conversation with someone more knowledgeable about the project.
+It's OK to ask for help.
+
+There's one key difference though, bringing advantage to you and other people in the future: 
+In almost all cases you should prefer the projects' official communication channels - this can be a mailing list, a chat room, the issue tracker or something similar depending on the purpose of having a more synchronous or asynchronous way of interacting or the varying needs for structure in the communication.
+All of those usually have in common that they are text-based, archived, searchable and linkable - this means your question and the answer will be written down.
+This way you could benefit from this passively documented knowledge on your search and will help future contributors to have the same advantage.
```

Comment:
I would mention that having the answers in writing also means that you can easily link to them in your official documentation should you later need them, instead of having to write them down again. 

See https://www.apache.org/dev/committers.html and search for "volunteeritis" on the page to find a link to a mail that was deemed important enough to be included in the new committer documentation.

Reply:
If we put in this extra explanation then I think that we should remove the following sentence.  Only need one additional sentence of elaboration, I'd say.

Reply:
Line 108 serves as an example for why the approach in line 106 might be great.
I would think that the extra example provide above is somewhat orthogonal to line 108 but don't have a real firm opinion here. What does @MaineC think here?  https://github.com/InnerSourceCommons/InnerSourceLearningPath/pull/117/files#r328171621